snuba/utils/streams/kafka.py

```diff
@@ -120,22 +121,85 @@ def close(self) -> None:
 DEFAULT_QUEUED_MIN_MESSAGES = 10000
 
 
-def build_kafka_consumer(
+def build_kafka_consumer_configuration(
     bootstrap_servers: Sequence[str],
     group_id: str,
     auto_offset_reset: str = "error",
     queued_max_messages_kbytes: int = DEFAULT_QUEUED_MAX_MESSAGE_KBYTES,
     queued_min_messages: int = DEFAULT_QUEUED_MIN_MESSAGES,
-) -> KafkaConsumer:
-    return KafkaConsumer(
-        {
-            "enable.auto.commit": False,
-            "bootstrap.servers": ",".join(bootstrap_servers),
-            "group.id": group_id,
-            "default.topic.config": {"auto.offset.reset": auto_offset_reset},
-            # overridden to reduce memory usage when there's a large backlog
-            "queued.max.messages.kbytes": queued_max_messages_kbytes,
-            "queued.min.messages": queued_min_messages,
-            "enable.partition.eof": False,
-        }
+) -> Mapping[str, Any]:
+    return {
+        "enable.auto.commit": False,
+        "bootstrap.servers": ",".join(bootstrap_servers),
+        "group.id": group_id,
+        "default.topic.config": {"auto.offset.reset": auto_offset_reset},
+        # overridden to reduce memory usage when there's a large backlog
+        "queued.max.messages.kbytes": queued_max_messages_kbytes,
+        "queued.min.messages": queued_min_messages,
+        "enable.partition.eof": False,
+    }
+
+
+class KafkaConsumerWithCommitLog(KafkaConsumer):
```

Comment:
I think that making this a subclass is a bit heavy handed, but is the most straightforward approach for now.

I think it'd be better to allow this to be attached to a commit callback, but we'd have to start dealing with multithreading since otherwise `producer.poll` would never be called and we'd never know if the callback is actually publishing or not. That adds a lot of complexity that is really not necessary here but I think would be a cleaner abstraction.

I think a lot of these method return values would be better fits for returning `Future` instances (that way the caller can choose whether to treat them as blocking or not), so I think we'll get there eventually but there's no requirement for it so I'm trying to keep it simple.